### ISSUE TITLE
Support root on iSCSI (non-HBA)

### ISSUE THREAD

**Issue by jlebon** [coreos/fedora-coreos-tracker#1590]
Some bare metal systems (and some cloud providers offering bare metal instance types, like [Oracle Cloud](https://github.com/coreos/fedora-coreos-tracker/issues/414)) run their boot disk off of iSCSI. In setups without an HBA, it's up to the OS to act as the iSCSI initiator. During the initramfs, the OS connects to the target to be able to mount the root device and keep going from there. There's a pile of code that already exists for this in dracut that we should be able to reuse (look for the `rd.iscsi` keys in https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html).

The main complexity for us is ensuring that it meshes well with our existing maze of initrd services and ordering.

**Comment by jbtrystram** [comment]
I did a bit of testing with iPXE and ibft. 
booting the following `boot.ipxe` : 
```
#!ipxe
sanboot iscsi:10.0.2.15::::iqn.2023-10.coreos.target.vm:core
```
Then adding `rd.iscsi.ibft=1 rd.iscsi.firmware=1` in kargs. 
In initramfs we have: 
```
:/root# iscsiadm -m fw
# BEGIN RECORD 6.2.1.4
iface.initiatorname = iqn.2010-04.org.ipxe:00000000-0000-0000-0000-000000000000
iface.transport_name = tcp
iface.hwaddress = 52:54:00:12:34:56
iface.bootproto = DHCP
iface.ipaddress = 10.0.3.15
iface.prefix_len = 24
iface.subnet_mask = 255.255.255.0
iface.gateway = 10.0.3.2
iface.primary_dns = 10.0.3.3
iface.vlan_id = 0
iface.net_ifacename = ens3
node.name = iqn.2023-10.coreos.target.vm:coreos
node.conn[0].address = 10.0.2.15
node.conn[0].port = 3260
node.boot_lun = 00000000
# END RECORD
```

This looks like it's pulled from `/sys/firmware/ibft`: 
```
:/root# cat /sys/firmware/ibft/target0/target-name 
iqn.2023-10.coreos.target.vm:coreos
:/root# cat /sys/firmware/ibft/target0/{ip-addr,port} 
10.0.2.15
3260
:/root# cat /sys/firmware/ibft/initiator/initiator-name 
iqn.2010-04.org.ipxe:00000000-0000-0000-0000-000000000000
```

**Comment by jlebon** [comment]
I opened https://github.com/coreos/fedora-coreos-config/pull/2702, with which I'm able to get FCOS up and running all the way to the real root on iSCSI via iPXE + `rd.iscsi.firmware=1`.

This addresses the generic case and isn't specific to Oracle Cloud. In theory, for Oracle Cloud enablement, all that's left is for AI to do the right wiring so that it adds `rd.iscsi.firmware=1` at install time. But we still need to verify that's all we need.

**Comment by jlebon** [comment]
Follow-up on Oracle Cloud enablement: I've successfully tested RHCOS with https://github.com/coreos/fedora-coreos-config/pull/2702 in Oracle Cloud on all the bare metal instance types [certified for RHEL](https://catalog.redhat.com/cloud/detail/216977). For good measure, I've also tested the VM instance types there, but those don't use iSCSI so it was expected to work.

As expected, one needs to add `rd.iscsi.firmware=1`, just like you would in traditional RHEL. I've also had to add `ip=ibft` so that NM only brings up networking on the iBFT device. Otherwise, it'll try to bring it up on all the NICs, some of which take much longer to get DHCP and can cause boot to timeout. (Of course, an alternative there is to bump the timeout though it seems much cleaner to only bring up the NIC we actually need to find the root disk.)
